fecgov/fec-cms
# Sort MUR by case_no

## Summary
Make MURS sortable by case number at:
https://www.fec.gov/data/legal/search/murs/ --OR--
https://www.fec.gov/data/legal/search/enforcement/  (defaults to MUR search)

- Resolves #5261

### Future work: 
This should work for AOs and possibly ADR/AFS. If we want to repurpose it for these, I recommend we remove the JS from the MURs template and revise the script to work for all of them and put it under /`js/pages` ,  `/js/modules` or `legal.js`. If that is possible, given the differences in those results-tables. The templates/views also would need to be edited like the MURs'  were for this PR.

### Required reviewers

One frontend, one backend, one UX

### Impacted areas of the application

modified:   fec/static/scss/components/_table-styles.scss
modified:   legal/templates/legal-search-results-murs.jinja
modified:   legal/templates/partials/legal-search-results-mur.jinja
modified:   legal/templates/partials/legal-pagination.jinja
modified:   legal/views.py

## Screenshots

![mur_sort](https://user-images.githubusercontent.com/5572856/217044232-080ae42c-eccf-4260-852e-99c048e39a0b.gif)


## Related PRs

Related PRs against other branches:

[#5246](https://github.com/fecgov/openFEC/pull/5246) Add sorting by case number and by advisory opinions number in legal/search

## How to test
- [ ] **Remove develop deploy task before merge**

- checkout `feature/5261-sort-murs-by-case-no` 
- `npm run build`
- `npm run test-single`
- `pytest` -AND- `cd ../ && pytest`
-  Go to http://127.0.0.1:8000/data/legal/search/murs/
- Test sort toggle on first column
- Apply some filters. Does it maintain the current sort order after applying filters?
- Test pagination with/without filters applied.
- Make sure the `aria` attributes on the sort toggle are correct upon changing sort direction
- Do we want the extra JS that maintains the position of the page when the page reloads (`see lines 115 and 120 of murs template` )? It has pros and cons -- Its great when using the toggle button, but maybe not as great when you are scrolled down to use the `Apply filters` button. 
     -  Also,  clicking pagination btns does not maintain page position because it loads the refreshed page to the `#results-murs` anchor. 
    - FYI... admittedly, the gif screenshot above implies that there is no visual page-jump on reload. While I was able to capture this ideal experience in a GIF for the PR, that's not always the experience since the page does have to reload to sort the data. :- {
